File: code/334.increasing-triplet-subsequence.php
<?php
/*
 * @lc app=leetcode id=334 lang=php
 *
 * [334] Increasing Triplet Subsequence
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $nums
     * @return Boolean
     */
    function increasingTriplet($nums) {
        // e.g.
        $length = count($nums);

        if ($length < 3) return false;

        $smallest = $secondSmallest = PHP_INT_MAX;

        foreach ($nums as $num) {
            if ($num <= $smallest) {
                $smallest = $num;
            } elseif ($num <= $secondSmallest) {
                $secondSmallest = $num;
            } else {
                return true;
            }
        }
        return false;
    }
}
// @lc code=end

File: code/334.increasing-triplet-subsequence.go
package code

import "math"

/*
 * @lc app=leetcode id=334 lang=golang
 *
 * [334] Increasing Triplet Subsequence
 */

// @lc code=start
func increasingTriplet(nums []int) bool {
	smallest, secondSmallest := math.MaxInt, math.MaxInt

	for _, v := range nums {
		if v <= smallest {
			smallest = v
		} else if v <= secondSmallest {
			secondSmallest = v
		} else {
			return true
		}
	}

	return false
}

// @lc code=end
